infrastructure/main.tf

```diff
@@ -16,20 +16,20 @@ data "azurerm_key_vault" "sscs_key_vault" {
 }
 
 data "azurerm_key_vault" "send_grid" {
-  provider = azurerm.send-grid
-
+  provider            = azurerm.send-grid
   name                = var.env != "prod" ? "sendgridnonprod" : "sendgridprod"
   resource_group_name = var.env != "prod" ? "SendGrid-nonprod" : "SendGrid-prod"
 }
 
 data "azurerm_key_vault_secret" "send_grid_api_key" {
-  provider = azurerm.send-grid
-
+  provider     = azurerm.send-grid
   key_vault_id = data.azurerm_key_vault.send_grid.id
   name         = "sscs-api-key"
 }
 
 resource "azurerm_key_vault_secret" "sendgrid_api_key" {
+  count = contains(["prod", "demo"], var.env) ? 1 : 0
```

Comment:
Is this going to add it for prod and demo?

Reply:
These already exist in prod and demo. They don't exist in the other envs. The 'count' section prevents the creation of that secret in ithc and other envs.